allensdk/brain_observatory/ecephys/ecephys_session.py

```diff
@@ -1124,7 +1126,22 @@ def _build_stimulus_presentations(
         # pandas groupby ops ignore nans, so we need a new "nonapplicable"
         # value that pandas does not recognize as null ...
         stimulus_presentations.replace("", nonapplicable, inplace=True)
-        stimulus_presentations.fillna(nonapplicable, inplace=True)
+
+        # pandas does not automatically convert boolean cols for fillna
+        boolean_colnames = stimulus_presentations.dtypes[
+            stimulus_presentations.dtypes == "boolean"].index
+        col_type_map = {colname: "object" for colname in boolean_colnames}
+        stimulus_presentations = stimulus_presentations.astype(
+            col_type_map).fillna(nonapplicable)
+
+        # eval str(numeric) and str(lists), convert lists to tuple for
+        # dict key compatibility
```

Comment:
The PR description above provides some example along with the rationale for calling eval. 
* stimulus_presentations['spatial_frequency'] contains numeric values of mixed str and float types, along with str(list)
  *Unique values from spatial_frequency column - current
`['0.08', '[0.0, 0.0]', '0.04', 0.08, 0.04, 0.32, 0.02, 0.16]`
  * [Test in static_gratings expects float vals for spatial_frequency (sfvals)](https://github.com/AllenInstitute/AllenSDK/blob/f69f0649428346a11461472f374205916756ff4c/allensdk/test/brain_observatory/ecephys/stimulus_analysis/test_static_gratings.py#L53)

Reply:
Thanks for adding a description to the PR. I also think an inline description would be helpful since the data structure is extremely messy and unconventional